### PR TITLE
Add doc for impl From for Error

### PR DESCRIPTION
**Pull request by phungleson**
As part of issue #51430 (cc @skade).

The impl is very simple, let me know if we need to go into any details.

### CODE REVIEW

**Comment by pietroalbini** [comment]
Ping from triage @skade! This PR needs your review.

**Comment by TimNN** [comment]
Ping from triage @skade / @rust-lang/docs: This PR requires your review.

File: src/libstd/io/error.rs [hunk 1]
```diff
@@ -212,6 +212,12 @@ impl ErrorKind {
 /// the heap (for normal construction via Error::new) is too costly.
 #[stable(feature = "io_error_from_errorkind", since = "1.14.0")]
 impl From<ErrorKind> for Error {
+    /// Converts a [`ErrorKind`] into a [`Error`].
+    ///
+    /// This conversion allocates a new error with simple repr.
```
Comment (GuillaumeGomez) [inline src/libstd/io/error.rs]: `repr`? (please write the full word).

**Comment by GuillaumeGomez** [comment]
Please add an example.

**Comment by phungleson** [comment]
Hey, I am currently traveling, will look at this again in around 3 weeks.

**Comment by GuillaumeGomez** [comment]
Please remove `fn main()` here and just keep its code.

**Comment by frewsxcv** [comment]
To expand on @GuillaumeGomez 's comment, you don't need to add the `main` function in documentation examples. See also:

https://doc.rust-lang.org/book/first-edition/documentation.html#documentation-as-tests

**Comment by phungleson** [comment]
Cool thanks @frewsxcv @GuillaumeGomez `main` has been removed

File: src/libstd/io/error.rs [hunk 2]
```diff
@@ -212,6 +212,19 @@ impl ErrorKind {
 /// the heap (for normal construction via Error::new) is too costly.
 #[stable(feature = "io_error_from_errorkind", since = "1.14.0")]
 impl From<ErrorKind> for Error {
+    /// Converts a [`ErrorKind`] into a [`Error`].
```
Comment (GuillaumeGomez) [inline src/libstd/io/error.rs]: Just thought: shouldn't it be "a**n** Error"?

**Comment by phungleson** [comment]
@frewsxcv not at all thanks!

**Pull request merged**